pkg/noun/nock.c

```diff
@@ -3191,3 +3207,64 @@ u3n_nock_an(u3_noun bus, u3_noun fol)
 
   return u3n_nock_et(gul, bus, fol);
 }
+
+/* u3n_sstack_init: initalize a root node on the spin stack 
+*/
+void
+u3n_sstack_init()
+{
+  c3_w shm_fd = shm_open(SLOW_STACK_NAME, O_CREAT | O_RDWR, 0666);
+  if ( -1 == shm_fd) {
+    perror("shm_open failed");
+    return;
+  }
+
+  if ( -1 == ftruncate(shm_fd, PSIZE)) {
+    perror("truncate failed");
+    return;
+  }
+
+  stk_u = mmap(NULL, PSIZE, PROT_READ | PROT_WRITE, MAP_SHARED, shm_fd, 0);
+  
+  if ( MAP_FAILED == stk_u ) {
+    perror("mmap failed");
+    return;
+  }
+
+  stk_u->off_w = 0;
+  //u3n_sstack_push(c3__root);
+}
+
+/* u3n_sstack_push: push a noun on the spin stack.
+*/
+void
+u3n_sstack_push(u3_noun nam)
+{
+  if ( c3n == u3ud(nam) ) {
+    return;
+  }
+
+  c3_w  met_w = u3r_met(3, nam);
+  
+  // Exit if full
+  if ( sizeof(stk_u->dat_y) < stk_u->off_w + met_w + sizeof(c3_w) ) {
+    return;
```

Comment:
If this happens we need a way to make sure that we don't pop unnecessarily.

Reply:
This is fixed by adding an extra member in the shared page to store the overflow count